app/stores/MessageUnreadStore.js

```diff
@@ -100,14 +111,18 @@ let UnreadStore = assign({}, BaseStore, {
                 break;
             case Constants.ChannelActionTypes.CHANGE_CHANNEL:
                 let backEndChannelId = payload.backEndChannelId;
+                var hasUnread = _unreadBool.get(backEndChannelId);
                 _unreadBool = _unreadBool.set(backEndChannelId, false);
-                UnreadStore.emit(IMConstants.EVENTS.CHANNEL_UNREAD_CHANGE_PREFIX + backEndChannelId, {unread : false});
+                _unreadCount = _unreadCount.set(backEndChannelId, 0);
+                UnreadStore.emit(IMConstants.EVENTS.CHANNEL_UNREAD_CHANGE_PREFIX + backEndChannelId, {unread : false, unreadCount : 0});
                 // cauz socket may not ready
+                if (!hasUnread)
```

Comment:
IMO: It works, but the code is a little bit dirty, you can put the logic inside the function below.
The reason is, if "hasUnread" is false, then we don't send "message:seen" event.
If so, we can put this logic inside the following function.

Or, if you have other consideration, tell me pls.